### PR TITLE
doc: Add document describing access controls at the project

### PR DESCRIPTION
**Pull request by sxa**
FAQ will be updated later to point at this, but since https://github.com/adoptium/infrastructure/pull/2977 is in-flight I'm holding off on that for now.

##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [x] other documentation is changed or added (if applicable)

Fixes https://github.com/adoptium/infrastructure/issues/2967 and https://github.com/adoptium/infrastructure/issues/2970

### CODE REVIEW

**Review by karianna — APPROVED** [review]

File: docs/AccessControl.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+## Jenkins build/test agents
+
+(At present, build and test agents have the same access controls on them but
+that will potentially change in the future as we look to lock down the build
+systems to a higher level)
+
+Administrative access to the build systems is currently granted to all members of the
+infrastructure team although in some cases individuals may have more
+restricted access (e.g.  to administer certain types of machine).  These
+policies are controlled by Bastillion which distributes the infrastructure
```
Comment (tellison) [inline docs/AccessControl.md:19]:
```suggestion
policies are controlled by [Bastillion](https://www.bastillion.io/) which distributes the infrastructure
```

File: docs/AccessControl.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+For Bastillion, the relevant groups are `root`, `root-marist`, `root-aix`,
+`root-equinix` and `jenkins`. Each infrastructure team user should generally be added
+to all of those.
+([Related issue](https://github.com/adoptium/infrastructure/issues/2970))
+If we implement stricter controls on the static build systems then it is
+likely we would have a `root-build` profile in Bastillion to manage those.
+
+**Onboarding process:** Add a User to Bastillion and add the user to required profiles
+
+**Offboarding process:**qs Remove the User from the required profiles, remove the account
```
Comment (tellison) [inline docs/AccessControl.md:34]:
```suggestion
**Offboarding process:** Remove the User from the required profiles, remove the account
```

File: docs/AccessControl.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+To request this, a user should create an issue in the infrastructure
+repository using [this
+template](https://github.com/adoptium/infrastructure/issues/new?assignees=sxa&labels=Temp+Infra+Access&template=machineaccess.md&title=Access+request+for+%3Cyour+username%3E)
+which will be labelled as "Temp Infra Access" and allows them to specify and
+justify the level of access.  In most cases this will be a normal user
+account which will be granted access via their ssh key.  In very rare cases
+access to jenkins or root may be provided, although bear in mind this will
+be very short lived as Bastillion will typically override that access.  In
+either case, the issue should be left open until the access is removed.
```
Comment (tellison) [inline docs/AccessControl.md:102]: I feel duty-bound to mention granting time-limited ssh access control again by key signing like the early days of AOJDK. I know I always say that! One day.

File: docs/AccessControl.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+**Onboarding** An Adoptium BitWarden admin needs to onboard the user into
+the Adoptium org, and grant access to the `infrastructure` Group (or
+`adopt-infra-admins` for some of the providers)
+
+**Offboarding** Remove access, change relevant passwords.
+
+## secrets / dotgpg
+
+We currently store various credentials in the provate secrets repository where the
```
Comment (tellison) [inline docs/AccessControl.md:144]:
```suggestion
We currently store various credentials in the private secrets repository where the
```

**Review by tellison — CHANGES_REQUESTED** [review]

**Comment by tellison** [comment]
Thanks for getting this written down

File: docs/AccessControl.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
+# Guide to access controls for infrastructure
+
+This document providers an overview of the access controls which are applied
+at different parts of the project in order to secure the systems and allow
+people to have access when required for onboarding and offboarding people. 
+Note that this document's scope does not cover the committer/contributor
+process for the Adoptium github org which is managed by Eclipse and
+controlled via their normal contributor election process.
```
[the remark this answers is not in the document]
Comment (sxa) [reply docs/AccessControl.md:8]: I'm tempted to reference our doc on interacting with eclipse, but that more direct one on that topic is good too.

File: docs/AccessControl.md (hunk 3 above)
Comment (sxa) [reply docs/AccessControl.md:102]: I actually have no recollection of that - do you have a reference? Feel free to raise an issue on it as I'm not sure how that was implemented.

Are you happy to approve this with the other changes? For some reason I'm not getting a "Re-request review" option for you on this.

**Review by tellison — APPROVED** [review]

**Pull request merged**